slack-api-model/src/main/java/com/slack/api/util/json/GsonFileFactory.java

```diff
@@ -0,0 +1,67 @@
+package com.slack.api.util.json;
+
+import com.google.gson.*;
+import com.slack.api.model.File;
+
+import java.lang.reflect.Type;
+
+public class GsonFileFactory implements JsonDeserializer<File>, JsonSerializer<File> {
+
+    // This is just a workaround to customize Gson library behavior
+    // You don't need to edit this class at all
```

Comment:
Not sure I understand this comment, do we still need it?

Reply:
As mentioned below, if you run `context.deserialize(jsonObject, File.class);` within the deserialize method, it results in StackOverflowError because the same method is associated with the File.class. To let gson mechansim to run a different auto-binding process without stack overflow, I've added this sub class. This is internal hidden class, so you can rename it whenever you want to do so. Also, if you find a better solution to achive the same, please feel free to change this part. It won't be a breaking change because this class and the the code using it are hidden from users.